### PR TITLE
Version 2.0.0

### LINKED ISSUE #276: Version 2.0 Release
This ticket is to track everything we're batching up in #275 for the 2.0 release:

- [x] Remove internal usage of mobx-react `Provider` #259 
- [x] Break `injectTreeNode` into composable parts (and kill autoAdd for now) #272
- [x] Kill Quick Start Story (for now - we'll bring it back later after 2.0 so we can think through it holistically instead of just patching it) #272
- [x] Context Based Theme API (main breaking change is wrapping all the component props into a single prop) #98
- [x] Kill mobx-react-utils #21
- [x] ~Clean up TermsStatsTable API #104~
- [ ] Optional Modal/Popover Lens API #317

### PR DESCRIPTION
**Pull request by daedalus28**
This PR is tracking the 2.0 release - as we merge things into v2, this PR should auto update 🎉 

Closes #276

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>:exclamation: This PR is BIG (+501 -500)  
Please keep it below 500 net changes</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>You should add at least 1 more reviewer to the PR</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Branch being merged does not follow [Git Flow](http://nvie.com/posts/a-successful-git-branching-model/)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 556ab16a7a2257fde126759d07ba4c64ead058cc
</p>

File: src/index.js [hunk 1]
```diff
@@ -1,10 +1,24 @@
-import QueryBuilder from './queryBuilder/'
+// utils
+export * from './utils/schema'
+export * from './utils/theme'
+
+// exampleTypes
+export * as ExampleTypes from './exampleTypes'
```
Comment (daedalus28) [inline src/index.js]: let's just 
```js
export * from './exampleTypes'
```
 😨

File: src/index.js [hunk 2]
```diff
@@ -1,10 +1,24 @@
...
-import ModalFilterAdder from './ModalFilterAdder'
-import FilterButtonList from './FilterButtonList'
+export FilterAdder from './FilterAdder'
+export FilterButtonList from './FilterButtonList'
+export SearchFilters, { SearchTree } from './SearchFilters'
+export SearchLayout from './SearchLayout'
+export ToggleFiltersHeader from './ToggleFiltersHeader'
+
+// component library
+export * from './greyVest'
```
Comment (daedalus28) [inline src/index.js]: let's not do this and instead publish greyVest as its own thing

File: src/index.js [hunk 3]
```diff
@@ -1,10 +1,24 @@
...
 
-export { QueryBuilder, FilterAdder, ModalFilterAdder, FilterButtonList }
+// themes
+export baseTheme from './themes/base'
```
Comment (daedalus28) [inline src/index.js]: Let's export a themes object which has these on it to enable this usage:

```js
import { ThemeProvider, themes } from 'contexture-react'

<ThemeProvider theme={themes.greyVest}>
//...
</ThemeProvider>
```

File: .storybook/config.js [hunk 4]
```diff
@@ -5,8 +5,10 @@ import 'babel-polyfill'
 setAddon(JSXAddon)
 
 function loadStories() {
-  const req = require.context('../stories', true, /\.stories\.js$/)
-  req.keys().forEach(filename => req(filename))
+  const stories = require.context('../stories', true, /\.stories\.js$/)
+  stories.keys().forEach(stories)
+  const src = require.context('../src', true, /\.stories\.js$/)
```
Comment (daedalus28) [inline .storybook/config.js]: See #326

File: src/exampleTypes/ResultTable.js [hunk 5]
```diff
@@ -132,7 +161,7 @@ let Header = withStateLens({ popover: false, adding: false, filtering: false })(
       hideMenu,
       typeDefault,
     } = fieldSchema
-    HeaderCell = fieldSchema.HeaderCell || HeaderCell
+    let HeaderCell = fieldSchema.HeaderCell || HeaderCellDefault
```
[the remark this answers is not in the document]
Comment (michsa) [reply src/exampleTypes/ResultTable.js]: I believe we decided this was safe to remove from ResultTable - I'll add it to the changelog

**Pull request merged**